lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -260,28 +236,20 @@ def test_unpack_with_exception(
         """Test to check the unpacking functionality of the FileTree when
         an Exception occured"""
         source_tarball, source_md5, md5 = tarball
-        tarball_name = source_tarball.name.removesuffix(".tar.xz")
 
         def mock_run(args, **kwargs):
-            assert args == [
-                "tar",
-                "-xf",
-                "--no-same-owner",
-                "--delay-directory-restore",
-                "--force-local",
-                (server_config.ARCHIVE / "ABC" / source_tarball.name),
-            ]
+            assert args[0] == "tar"
             raise subprocess.TimeoutExpired("tar", 43)
 
         monkeypatch.setattr("subprocess.run", mock_run)
         tree = FileTree(server_config, make_logger)
         tree.create("ABC", source_tarball)
-
+        tarball = tree.find_dataset(md5)
         with pytest.raises(TarballUnpackError) as exc:
             tree.unpack(md5)
         assert (
             str(exc.value)
-            == f"An error occurred while unpacking {tarball_name}: Command 'tar' timed out after 43 seconds"
+            == f"An error occurred while unpacking {tarball.tarball_path}: Command 'tar' timed out after 43 seconds"
         )
```

Comment:
Shouldn't the test code already "know" the tarball path without having to use `tree.find_dataset()` to look it up?  

(Having the test depend on the CUT like this seems sort of circular:  "the CUT must return the name that the CUT says is the name!"  I suppose that that tests for consistency, which is good, but it doesn't detect damage or confusion....)

Ditto for the subsequent tests.